go/cmd/dolt/commands/log.go

```diff
@@ -110,576 +88,232 @@ func (cmd LogCmd) ArgParser() *argparser.ArgParser {
-	if ok {
-		if len(opts.excludingCommitSpecs) > 0 {
-			return nil, fmt.Errorf("cannot use --not argument with two dots or ref with ^")
-		}
-		if len(opts.tableName) > 0 {
-			return nil, fmt.Errorf("cannot use --not argument with table")
-		}
-		for _, excludingRef := range excludingRefs {
-			notCs, err := doltdb.NewCommitSpec(excludingRef)
-			if err != nil {
-				return nil, fmt.Errorf("invalid commit %s\n", excludingRef)
-			}
-
-			opts.excludingCommitSpecs = append(opts.excludingCommitSpecs, notCs)
-		}
+	logRows, err := GetRowsForSql(queryist, sqlCtx, query)
+	if err != nil {
+		return handleErrAndExit(err)
 	}
 
-	return opts, nil
+	return logCommits(apr, logRows, queryist, sqlCtx)
 }
 
-func (opts *logOpts) parseRefsAndTable(ctx context.Context, apr *argparser.ArgParseResults, dEnv *env.DoltEnv) error {
-	// `dolt log`
-	if apr.NArg() == 0 {
-		return nil
-	}
-
-	if strings.Contains(apr.Arg(0), "..") {
-		if apr.NArg() > 1 {
-			return fmt.Errorf("Cannot use two or three dot syntax when 2 or more arguments provided")
-		}
-
-		// `dolt log <ref>...<ref>`
-		if strings.Contains(apr.Arg(0), "...") {
-			refs := strings.Split(apr.Arg(0), "...")
+// constructInterpolatedDoltLogQuery generates the sql query necessary to call the DOLT_LOG() function.
+// Also interpolates this query to prevent sql injection.
+func constructInterpolatedDoltLogQuery(apr *argparser.ArgParseResults, queryist cli.Queryist, sqlCtx *sql.Context) (string, error) {
+	var params []interface{}
 
-			for _, ref := range refs {
-				cs, err := getCommitSpec(ref)
-				if err != nil {
-					return err
-				}
-				opts.commitSpecs = append(opts.commitSpecs, cs)
-			}
+	var buffer bytes.Buffer
+	var first bool
+	first = true
 
-			mergeBase, verr := getMergeBaseFromStrings(ctx, dEnv, refs[0], refs[1])
-			if verr != nil {
-				return verr
-			}
-			notCs, err := getCommitSpec(mergeBase)
-			if err != nil {
-				return err
-			}
-			opts.excludingCommitSpecs = append(opts.excludingCommitSpecs, notCs)
+	buffer.WriteString("select commit_hash from dolt_log(")
 
-			return nil
+	writeToBuffer := func(s string) {
+		if !first {
+			buffer.WriteString(", ")
 		}
+		buffer.WriteString(s)
+		first = false
+	}
 
-		// `dolt log <ref>..<ref>`
-		refs := strings.Split(apr.Arg(0), "..")
-		notCs, err := getCommitSpec(refs[0])
-		if err != nil {
-			return err
+	if apr.PositionalArgsSeparatorIndex >= 0 {
+		for i := 0; i < apr.PositionalArgsSeparatorIndex; i++ {
+			writeToBuffer("?")
+			params = append(params, apr.Arg(i))
 		}
-
-		cs, err := getCommitSpec(refs[1])
+		tableNames := ""
+		for i := apr.PositionalArgsSeparatorIndex; i < apr.NArg(); i++ {
+			tableNames = tableNames + apr.Arg(i) + ","
+		}
+		if tableNames != "" {
+			tableNames = strings.TrimSuffix(tableNames, ",")
+			params = append(params, tableNames)
+			writeToBuffer("'--tables'")
+			writeToBuffer("?")
+		}
+	} else {
+		rows, err := GetRowsForSql(queryist, sqlCtx, "show tables")
```

Comment:
`show tables` will give you the table on the current HEAD. If you are logging on a different branch, this will possibly give incorrect results. This would be a really good test to add.